File: LinkedList/LinkedList-Csharp/src/Link.cs
namespace Data
{
  class Link
  {
    public Link()
    {
      this.top=null;
      this.bottom=null;
      this.size=0;
    }

    public void addBottom(string data)
    {
      Node newNode=new Node(new Data(data));

      if(isEmpty())
      {

      }
    }

    private Node top,bottom;
    private int size;
  }
}

File: LinkedList/Doubly/src/Link/Link.java
package Link;

public class Link 
{
  public void push_rear(String data)
  {
    Node node=new Node(data);
    if(this.isEmpty())
    {
      this.front=node;
      this.rear=node;
    }
  }

  public void push_front(String data)
  {
    Node node=new Node(data);
    if(this.isEmpty())
    {
      this.front=node;
      this.rear=node;
    }
    else 
    {
      this.front.prev=node;
      node.next=this.front;
      this.front=node;
    }
    this.size++;
  }

  private boolean isEmpty(){return this.front==null;}
  private int size;
  private Node rear,front;

  public Link()
  {
    this.front=null;
    this.rear=null;
    this.size=0;
  }
}
